autoblocks/_impl/prompts/manager.py

```diff
@@ -111,18 +156,61 @@ async def _get_prompt(
         resp.raise_for_status()
         return Prompt.model_validate(resp.json())
 
-    async def _init_async(self, timeout: timedelta) -> None:
+    async def _set_prompt_snapshot(self, snapshot_id: str) -> None:
+        """
+        If this prompt has a snapshot set, use the /override endpoint to check if the
+        major version this prompt manager is configured to use is compatible to be
+        overridden with the snapshot.
+        """
+        # Double check we're in a testing context
+        if not is_testing_context():
+            log.error("Can't set prompt snapshot unless in a testing context.")
+            return None
+
+        resp = await global_state.http_client().post(
+            self._make_snapshot_override_request_url(snapshot_id),
+            timeout=self._init_timeout.total_seconds(),
+            headers={"Authorization": f"Bearer {self._api_key}"},
+            json=dict(
+                majorVersion=self.__prompt_major_version__,
+            ),
+        )
+
+        if resp.status_code == HTTPStatus.CONFLICT:
+            # The endpoint returns this status code when the snapshot is
+            # not compatible with the major version this prompt manager
+            # is configured to use.
+            raise IncompatiblePromptSnapshotError(
+                f"Can't override '{self._name}' with prompt snapshot '{snapshot_id}' because it is not compatible "
+                f"with major version '{self.__prompt_major_version__}'."
+            )
```

Comment:
i'm wondering how we can easily surface this in the UI, most likely the people that triggered this error aren't going to look at the logs of the run to know what happened.

Reply:
yeah really good point, i have a few ideas. i'll create a ticket and put thoughts there